src/main/java/teammates/ui/controller/InstructorCourseDetailsPageData.java

```diff
@@ -8,23 +8,165 @@
+                                             Const.Tooltips.COURSE_STUDENT_EDIT, "tooltip", null, isDisabled);
+        
+        isDisabled = !currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+        ElementTag sendInviteButton = createButton("Send Invite", "btn btn-default btn-xs", null, 
+                                                   getCourseStudentRemindLink(student), Const.Tooltips.COURSE_STUDENT_REMIND, 
+                                                   "tooltip", "return toggleSendRegistrationKey()", isDisabled);
+   
+        isDisabled = !currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+        String onClick = "return toggleDeleteStudentConfirmation('" + sanitizeForJs(student.course) 
+                         + "','" + sanitizeForJs(student.name) + "')";
+        ElementTag deleteButton = createButton("Delete", "btn btn-default btn-xs", null, getCourseStudentDeleteLink(student),
+                                               Const.Tooltips.COURSE_STUDENT_DELETE, "tooltip", onClick, isDisabled);
+        
+        ElementTag allRecordsButton = createButton("All Records", "btn btn-default btn-xs", null, 
+                                                   getStudentRecordsLink(student), Const.Tooltips.COURSE_STUDENT_RECORDS, 
+                                                   "tooltip", null, false);
 
-    public InstructorAttributes currentInstructor;
-    public CourseDetailsBundle courseDetails;
-    public List<StudentAttributes> students;
-    public List<InstructorAttributes> instructors;
-    public String studentListHtmlTableAsString;
+        row.getActions().add(viewButton);
+        row.getActions().add(editButton);
+        if (!student.isRegistered()) {
+            row.getActions().add(sendInviteButton);
+        }
+        row.getActions().add(deleteButton);
+        row.getActions().add(allRecordsButton);
+        
+        isDisabled = !currentInstructor.isAllowedForPrivilege(student.section, 
+                                            Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+        ElementTag addCommentButton = createButton("Add Comment", "btn btn-default btn-xs cursor-default", 
+                                                   null, "javascript:;", Const.Tooltips.COURSE_STUDENT_COMMENT, 
+                                                   "tooltip", null, isDisabled);
+        
+        isDisabled = !currentInstructor.isAllowedForPrivilege(student.section, 
+                                            Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+        
+        String content = "<span class=\"caret\"></span><span class=\"sr-only\">Add comments</span>";
+        ElementTag addCommentsButton = createButton(content, "btn btn-default btn-xs dropdown-toggle", null, 
```

Comment:
This may be better named `addCommentsDropdownButton` because only the arrow is really shown on the user's side.